model/src/pyrenew/process/differencedprocess.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+
+from __future__ import annotations
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+from numpyro.contrib.control_flow import scan
+from pyrenew.metaclass import RandomVariable, SampledValue
+
+
+class DifferencedProcess(RandomVariable):
+    """
+    Class for differenced stochastic process X(t),
+    constructed by placing a fundamental stochastic
+    process on the first differences (rates of change).
+    """
+
+    def __init__(
+        self,
+        name: str,
+        fundamental_process: RandomVariable,
+        differencing_order: int,
+        **kwargs,
+    ) -> None:
+        """
+        Default constructor
+
+        Parameters
+        ----------
+        name : str
+            Name of the stochastic process
+        fundamental_process : RandomVariable
+            Stochastic process for the
+            first differences. Should accept an
+            `n` argument specifying the number
+            of samples to draw.
+        differencing_order : int
+            How many fold-differencing the
+            the process represents. Must be
+            an integer greater than or
+            equal to 1. 1 represents a process
+            on the first differences (the rate
+            of change), 2 a process on the
+            2nd differences (rate of change of
+            the rate of change), et cetera.
+        **kwargs :
+            Additional keyword arguments passed to
+            the parent class constructor.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        The order of differencing is the discrete
+        analogue of the order of a derivative in single
+        variable calculus. A first difference (derivative)
+        represents a rate of change. A second difference
+        (derivative) represents the rate of change of that
+        rate of change, et cetera.
+        """
+        self.assert_valid_differencing_order(differencing_order)
+        self.differencing_order = differencing_order
+        self.fundamental_process = fundamental_process
+        super().__init__(name=name, **kwargs)
+
+    def integrate(
+        self, init_diff_vals: ArrayLike, highest_order_diff_vals: ArrayLike
+    ):
+        """
+        Integrate (de-difference) the differenced process,
+        obtaining the process values X(t=0), X(t=1), ... X(t)
+        from the n-th differences and a set of initial process /
+        difference values X(t=0), X^1(t=1), X^2(t=2), ...
+        X^(n-1)(t=n-1), where X^k(t) is the value of the n-th
+        differenc at index t of the process.
```

Comment:
```suggestion
        difference at index t of the process.
```

Reply:
Better to use math formatting here.

Reply:
Some precise definition of the nth difference could be helpful.

Reply:
Will change.

Reply:
addressed in 5bceb27